### ISSUE TITLE
speed up the build

### ISSUE THREAD

**Issue by afeld** [18F/jekyll_pages_api#20]
@mbland noted that the plugin might be accounting for not-insignificant slowdown, especially in sites with a lot of pages. TODO to look into some optimization.

**Comment by willnewby** [comment]
Is this still in need of attention?

**Comment by afeld** [comment]
Yes! Might be able to leverage some of the incremental build stuff in Jekyll 3, too.

**Comment by willnewby** [comment]
Neat. Alrighty. I'm not much of a Rubyist...yet. This seems as good a place
as any to start.

I'll check into the incremental build stuff and do some profiling.


---------------
Will Newby
Freelance SysAdmin/Coder/Brain
Phone: 612.208.3806
Email: willnewby@gmail.com

**Comment by afeld** [comment]
Awesome! Happy to talk through stuff if you like. Also, happy to keep this Jekyll 3-only if that makes things easier.

**Comment by willnewby** [comment]
Cool deal. 

Is there an IRC channel or slack somewhere that I missed?

**Comment by afeld** [comment]
@willnewby Good question – hadn't thought about it! Let's try https://gitter.im/18F/jekyll_pages_api in the short term.
